apps/public/public.go

```diff
@@ -251,3 +251,11 @@ func (pca Client) AcquireTokenByAuthCode(ctx context.Context, scopes []string, o
 func (pca Client) Accounts() []Account {
 	return pca.base.Accounts()
 }
+
+// AcquireTokenInteractive acquires a security token from the authority using the default web browser to select the account.
+func (pca Client) AcquireTokenInteractive(ctx context.Context, scopes []string) (AuthResult, error) {
```

Comment:
@abhidnya13 @keegan-caruso Porting question from previous PR about automated testing thing.

In `azidentity` we have a test hook that overrides `browser.Open` with a simulated response.  This would be half the solution, still need to mock the other API calls.

Reply:
I've added the hook so we can simulate the auth code request portion of the flow, and stubbed out a test for this.

There is no fake `oauth.Client` exposed at present (you can't even publicly override it) so I can't mock the token request part.  Poking around the code, there are bits and pieces (e.g. `fakeAccessTokens`) but nothing cohesive (unless I missed it).

Reply:
Thinking about this a little, maybe this is one ETOE that we put in another place where we can do overrides.  We definitely don't want to expose the oauth.Client to the outside.

Reply:
While I can support not exposing `oauth.Client`, without having some kind of fake we can't test this (or other methods) end-to-end (I assume that's what ETOE means).  Did you have something else in mind?